### PR TITLE
Update mapping code and sam simulation model for compression

### PR DESCRIPTION
**Pull request by bobcheng15**
Added the behavior simulation code of crddrop in compression mode required by the sparse unit test in lake. Also updated hw_node/merge_node.py to account for the additional configuration register cmrg_mode of crddrop.

### CODE REVIEW

**Comment by weiya711** [comment]
@kalhankoul96 do not approve it unless unit tests are also added for the ValDropper in the `sam/sim/test/primitives` folder

**Review by kalhankoul96 — APPROVED** [review]
LGTM

**Pull request merged**
